Duplicati/CommandLine/RecoveryTool/Restore.cs

```diff
@@ -171,7 +171,8 @@ public static int Run(List<string> args, Dictionary<string, string> options, Lib
                                 {
                                     if (f.BlocklistHashes == null)
                                     {
-                                        lookup.WriteHash(sw, f.Hash);
+                                        if (f.Size > 0)

```

Comment:
How can size be anything other than 0 if there are no hashes?

Reply:
I'm not sure how quickly review comments can be processed. As explained elsewhere, there's a still-broken Canary to fix. 

Fortunately I know a bit about this one. See #4884 cited above for some history, and see [Processing a large file](https://duplicati.readthedocs.io/en/latest/appendix-a-how-the-backup-process-works/#processing-a-large-file) for how blocklists are only used for files with more than 1 block. [Processing a small file](https://duplicati.readthedocs.io/en/latest/appendix-a-how-the-backup-process-works/#processing-a-small-file) only has a size and a hash, but the history cited in the issue explains how a zero length block isn't in the backup files, though its hash might make one think that...

You can see the large file blocklist expansion in the else below the code in question, which handles the small file case, and presumably relies on its file open (without data written from a block) to make the empty file. That's my impression of this.

[Database rebuild](https://forum.duplicati.com/t/database-rebuild/13884/7?u=ts678) was my attempt as part of a team effort to summarize how things fit together. There are lots of hashes, and a blocklist is a block, and is known by its hash, and the filelist.json in the dlist file will have one or more for a large file.